src/main/java/core/basesyntax/JavaStreamApi.java

```diff
@@ -29,16 +37,22 @@ public Long elementCount(List<String> elements, String element) {
      * Вернуть Optional первого элемента коллекции</p>
      **/
     public Optional<String> firstElement(List<String> elements) {
-        return null;
+
+        return elements.stream()
+                .findFirst();
     }
 
     /**
      * <p>4. Дана коллекция строк List of String elements
-     * (приметр: Arrays.asList(«a1», «a2», «a3», «a1»)).
+     * (пример: Arrays.asList(«a1», «a2», «a3», «a1»)).
      * Найти элемент в коллекции равный `element` или кинуть ошибку NoSuchElementException</p>
      **/
-    public String findElement(List<String> elements, String element) {
-        return null;
+    public String findElement(List<String> elements, String element) throws NoSuchElementException {
+
+        return elements.stream()
+                .filter(element::equals)
+                .findAny()
+                .orElseThrow(NoSuchElementException::new);
```

Comment:
```suggestion
                .orElseThrow();
```
this solution will be right as well
FYI